src/ray/core_worker/core_worker.cc

```diff
@@ -1811,6 +1821,51 @@ void CoreWorker::HandlePushTask(const rpc::PushTaskRequest &request,
   });
 }
 
+void CoreWorker::HandleStealWork(const rpc::StealWorkRequest &request,
+                                 rpc::StealWorkReply *reply,
+                                 rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(DEBUG) << "Entering HandleStealWork!";
+  // get maximum number of tasks to steal
+  int max_tasks_to_steal = request.max_tasks_to_steal();
+  RAY_CHECK(max_tasks_to_steal >= 0);
+
+  {
+    absl::MutexLock lock(&direct_task_receiver_->mu_);
+
+    int half = direct_task_receiver_->tasks_received_.size() / 2;
```

Comment:
Just made the change from `int` to `size_t`. When it comes to the choice of stealing _half_ of the task (instead of a larger or smaller fraction of the total), the idea is that when a _thief_ initiates stealing, it has no tasks in flight. So by stealing half of the tasks, we make sure that _thief_ and _victim_ end up with the same number of tasks in flight. Of course, if tasks are very short, when the _thief_ receives the stolen tasks, the _victim_ might have less than half of the original workload (because it has executed some in the meantime).